src/common/transformations/include/ngraph_ops/nms_ie_internal.hpp

```diff
@@ -40,6 +62,7 @@ class TRANSFORMATIONS_API NonMaxSuppressionIEInternal : public Op {
                                 const Output<Node>& soft_nms_sigma,
                                 int center_point_box,
                                 bool sort_result_descending,
+                                bool soft_nms,
```

Comment:
Why we explicitly add soft_nms attribute? It should be inferred from the value in `const Output<Node>& soft_nms_sigma`. In some cases if the value changed from zero to non-zero, or non-zero to zero, this attribute is not valid any more and maybe introduce incorrect result. It should be determined in each infer time and remove the assumption of const soft_nms_sigma input to cover more cases. What do you think?

Reply:
Because there is different logic in NMS-5 and NMS-9 even if soft_nms_sigma is positive, according the documentation in nms5:
func(iou) = 1 if iou <= iou_threshold else 0 when soft_nms_sigma == 0, else func(iou) = exp(-0.5 \* iou \* iou / soft_nms_sigma) if iou <= iou_threshold **else 0.**

in nms9:
 func(iou) = 1 if iou <= iou_threshold else 0 when soft_nms_sigma == 0, else func(iou) = exp(-0.5 \* iou \* iou / soft_nms_sigma).


The same difference you can see in the reference implementation, please take a look here:
https://github.com/openvinotoolkit/openvino/blob/master/src/core/reference/src/runtime/reference/non_max_suppression.cpp#L126

and 
https://github.com/openvinotoolkit/openvino/blob/master/src/core/reference/src/runtime/reference/non_max_suppression.cpp#L264
https://github.com/openvinotoolkit/openvino/blob/master/src/core/reference/src/runtime/reference/non_max_suppression.cpp#L315

So to support both NMS-5 and NMS-9 we need to add this parameter, in another case we can support only one NMS version

Reply:
soft mode in NMS-5 is not used in any frameworks and not an reasonable behavior. It should be totally replaced with nms-9. Ideally we should remove NMS-5 in OV in some stage when all update to NMS-9. You can refer here why soft-nms is introduced. In short, in soft-nms, the candidate boxed should never suppressed by IOU, just suppressed the decayed score by score threshold.

Reply:
so you mean leave this parameter and just extend the comment that it should be removed in future, right?

Reply:
I mean remove "bool soft_nms" from constructor, and deduce it in infer time, where we know the exact real-time value in input "soft_nms_sigma" tensor.

Reply:
I mean NMS-5 to NmsInternal will be incorrect transformation in such case

Reply:
in other words, if it's acceptable to remove NMS-5 support at all, I'll remove this logic. (@wilson-seok  what do you think?)